apps/studio/components/interfaces/Auth/AdvancedAuthSettingsForm/AdvancedAuthSettingsForm.tsx

```diff
@@ -67,20 +80,81 @@ const AdvancedAuthSettingsForm = () => {
 
   const isTeamsEnterprisePlan =
     isSuccessSubscription && subscription.plan.id !== 'free' && subscription.plan.id !== 'pro'
+  const isProPlanAndUp = isSuccessSubscription && subscription?.plan?.id !== 'free'
+  const promptProPlanUpgrade = IS_PLATFORM && !isProPlanAndUp
+
+  const hasPurchasedAuthMFAAddOn =
+    subscription?.project_addons &&
+    subscription.project_addons.length > 0 &&
+    subscription.project_addons[0].addons.some((addon) => addon.type === 'auth_mfa_phone')
   const promptTeamsEnterpriseUpgrade = IS_PLATFORM && !isTeamsEnterprisePlan
 
   const INITIAL_VALUES = {
     SITE_URL: authConfig?.SITE_URL,
     JWT_EXP: authConfig?.JWT_EXP,
     REFRESH_TOKEN_ROTATION_ENABLED: authConfig?.REFRESH_TOKEN_ROTATION_ENABLED || false,
+    // TOTP is enabled by default
+    MFA_PHONE_OTP_LENGTH: authConfig?.MFA_PHONE_OTP_LENGTH || 6,
+    MFA_PHONE_TEMPLATE: authConfig?.MFA_PHONE_SMS_TEMPLATE || 'Your code is {{ .Code }}',
     SECURITY_REFRESH_TOKEN_REUSE_INTERVAL: authConfig?.SECURITY_REFRESH_TOKEN_REUSE_INTERVAL,
     MFA_MAX_ENROLLED_FACTORS: authConfig?.MFA_MAX_ENROLLED_FACTORS || 10,
     DB_MAX_POOL_SIZE: authConfig?.DB_MAX_POOL_SIZE || 10,
     API_MAX_REQUEST_DURATION: authConfig?.API_MAX_REQUEST_DURATION || 10,
+    MFA_TOTP:
+      determineMFAStatus(
+        authConfig?.MFA_TOTP_VERIFY_ENABLED ?? true,
+        authConfig?.MFA_TOTP_ENROLL_ENABLED ?? true
+      ) || 'Enabled',
+    MFA_PHONE:
+      determineMFAStatus(
+        authConfig?.MFA_PHONE_VERIFY_ENABLED || false,
+        authConfig?.MFA_PHONE_ENROLL_ENABLED || false
+      ) || 'Disabled',
   }
 
+  function MfaStatusToState(status) {
+    return status === 'Enabled'
+      ? { verifyEnabled: true, enrollEnabled: true }
+      : status === 'Verify Enabled'
+      ? { verifyEnabled: true, enrollEnabled: false }
+      : { verifyEnabled: false, enrollEnabled: false }
+  }
+
+  const MFAFactorSelectionOptions = [
+    {
+      label: 'Enabled',
+      value: 'Enabled',
+    },
+    {
+      label: 'Verify Enabled',
+      value: 'Verify Enabled',
+    },
+    {
+      label: 'Disabled',
+      value: 'Disabled',
+    },
+  ]
+
   const onSubmit = (values: any, { resetForm }: any) => {
-    const payload = { ...values }
+    let payload = { ...values }
+    const { verifyEnabled: MFA_TOTP_VERIFY_ENABLED, enrollEnabled: MFA_TOTP_ENROLL_ENABLED } =
+      MfaStatusToState(values.MFA_TOTP)
+    if (isProPlanAndUp) {
+      const { verifyEnabled: MFA_PHONE_VERIFY_ENABLED, enrollEnabled: MFA_PHONE_ENROLL_ENABLED } =
+        MfaStatusToState(values.MFA_PHONE)
+      payload = {
```

Comment:
This logic looks suspicious. You're setting two values on Pro plans which then can get overwritten in line 151.

Reply:
Not too clear here - do expanding more on how they might get overwritten?  The intention was that these two lines would set `MFA_PHONE_ENROLL_ENABLED` and `MFA_PHONE_VERIFY_ENABLED` as they're only applicable on the Pro plan and up

```
payload = {
    ...payload,
        MFA_PHONE_ENROLL_ENABLED,
        MFA_PHONE_VERIFY_ENABLED,
      }
    }
```

while the lines on 151:

```
payload = {
      ...payload,
      MFA_TOTP_ENROLL_ENABLED,
      MFA_TOTP_VERIFY_ENABLED,
    }
```
Would set the TOTP variables as they're available on all plans.

Let me know if there's something I've missed though

Reply:
Ah, they're different values, I undestand now.